### DISCUSSION TITLE
Returns Optional's in DAO's/Repositories

### DISCUSSION THREAD

**Discussion by mzudse** [remsfal/remsfal-backend#106] [Ideas]
We could return a Optional<T> instead of a object directly. Then we could remove the null checks and improve the code readability.

```

userEntity.isPresent() // userEntity.isEmpty()
vs
userEntity != null // userEntity == null

```

```
public UserEntity findByEmail(final String email) {
    return find("email", email).singleResult();
}

public Optional<UserEntity> findByEmail(final String email) {
    return Optional.ofNullable(find("email", email).singleResult());
}
```

The PanacheRepositoryBase already contains the method 'findByIdOptional' as an alternative of 'findById'

**Comment by astanik** [comment]
This is a great idea, if we only talk about the return types of repositories. This would only affect the null checks in the controller layer.

This discussion can be implemented with https://github.com/remsfal/remsfal-backend/issues/90
